### PR TITLE
Revise description of SRTP

### LINKED ISSUE #71: 5.1: tcpcrypt and ZRTP
Comment from Ekr's review of -10:

5.1.  Pre-Connection Interfaces

[…]
Also, I'm not sure you want to omit tcpcrypt here, in that it does
have a way to do external authentication by exporting a session ID.
Also ZRTP allows for some sort of fingerprint-based auth.

tcpcrypt (and I believe ZRTP) both have session caches.

### PR DESCRIPTION
**Pull request by csperkins**
This revises the description of SRTP and the relation between SRTP, DTLS, and ZRTP.

Closes #71

### CODE REVIEW

File: draft-ietf-taps-transport-security.md [hunk 1]
```diff
@@ -418,7 +419,9 @@ like TCP.
 
 Application Payload Security Protocols:
 
-- SRTP {{?RFC7201}}
+- DTLS when used as a handshake protocol for SRTP {{?RFC7850}}
+- ZRTP {{?RFC4571}}
```
Comment (renghardt) [inline draft-ietf-taps-transport-security.md:423]: Should the reference here be RFC6189, as we use that reference for ZRTP above?

File: draft-ietf-taps-transport-security.md [hunk 2]
```diff
@@ -418,7 +419,9 @@ like TCP.
 
 Application Payload Security Protocols:
 
-- SRTP {{?RFC7201}}
```
Comment (renghardt) [inline draft-ietf-taps-transport-security.md:421]: This deletes the only reference to RFC 7201 from our document.
Is it intentional to drop this reference and use RFC4571 instead here?

**Review by renghardt — COMMENTED** [review]
Thank you for separating these protocols out!

The text in Section 3.2 sounds good to me, but if we're getting rid of the headline "ZRTP for Media Path Key Agreement", we will only have one subsection (3.2.1) within Section 3.2. So maybe the headline should stay?

Also, just want to double-check that the changed references are intentional, e.g., removing the reference to RFC 7201.

For the application interfaces, this basically says that SRTP doesn't have any of these interfaces on its own, but ZRTP has them, right?
Once this PR is merged, I'll change the summary table accordingly. As is, this means the table will just have a row for ZRTP, as I don't think having a separate empty row for SRTP makes sense.

File: draft-ietf-taps-transport-security.md (hunk 1 above)
Comment (csperkins) [reply draft-ietf-taps-transport-security.md:423]: RFC4571 describes the framing of RTP-related protocols over connection oriented transports. It maybe needs both references.

**Comment by csperkins** [comment]
> Thank you for separating these protocols out!
> 
> The text in Section 3.2 sounds good to me, but if we're getting rid of the headline "ZRTP for Media Path Key Agreement", we will only have one subsection (3.2.1) within Section 3.2. So maybe the headline should stay?

If we keep that headline, we need one for DTLS-SRTP too.

> Also, just want to double-check that the changed references are intentional, e.g., removing the reference to RFC 7201.

Yes, that was intentional.

> For the application interfaces, this basically says that SRTP doesn't have any of these interfaces on its own, but ZRTP has them, right?
> Once this PR is merged, I'll change the summary table accordingly. As is, this means the table will just have a row for ZRTP, as I don't think having a separate empty row for SRTP makes sense.

SRTP is just the record protocol. The key management is usually done by DTLS or occasionally by ZRTP.

**Review by tfpauly — APPROVED** [review]

**Review by renghardt — APPROVED** [review]
Thanks for the reply. 
I'm okay with merging these changes as they are, or you can add that one reference for ZRTP to have both.

**Pull request merged**